orchestrator/workflow.py

```diff
@@ -270,9 +287,16 @@ def wrapper(state: State) -> Process:
     return decorator
 
 
-def inputstep(name: str, assignee: Assignee) -> Callable[[InputStepFunc], Step]:
+def inputstep(
+    name: str,
+    assignee: Assignee,
+    resume_auth_callback: Authorizer | None = None,
```

Comment:
Maybe it's an option to use default callbacks instead of None, so you can always call the function and don't have to check if they are provided.

Reply:
I don't think this would work with the design, which hinges on being able to detect that `resume_auth_callback` is not provided - in which case the "normal" auth_callback is used

Reply:
I think that's a very good general approach that may not fit here without additional refactoring.

The callbacks for `workflow` ultimately use a default callback of `allow`, but for `inputstep` the behavior of "no callback given" is "look for a callback from the preceding `workflow` or `inputstep`." For this to work, we need a sentinel value on the resulting `Step` that can be examined at runtime.

That sentinel value doesn't have to be `None`, but I don't think it's ideal for it to be of type `Authorizer` or even callable if it shouldn't be called in the first place.